apps/web-connect/src/features/checkout/components/PromoCodeRow.tsx

```diff
@@ -21,6 +21,16 @@ export function PromoCodeRow() {
 
     const [promo, setPromo] = useState(false);
 
+    useEffect(() => {
+        const queryString = window.location.search;
+        const queryParams = new URLSearchParams(queryString);
+        const prefilledPromoCode = queryParams.get("promoCode");
+
+        if (prefilledPromoCode) {
+            setPromo(true);
+        }
+    }, []);
```

Comment:
Does this now exist in both places Checkout.tsx and the PromoCodeRow ?

Not a required change, just want to confirm.

It would make sense considering the context set in Checkout and then reloaded in the PromoCode component.
At one point we should refactor this if feasable.

Reply:
Yes, though I opted to make this simple fix instead of making a more complex solution like passing down props or adding another state variable to the giant context object.